tool/tsh/common/tsh.go

```diff
@@ -486,6 +489,14 @@ func (c *CLIConf) Stdin() io.Reader {
 	return os.Stdin
 }
 
+// Getenv retrieves the value of the environment variable named by the key.
+func (c *CLIConf) Getenv(key string) string {
+	if c.envMap != nil {
+		return c.envMap[key]
+	}
+	return os.Getenv(key)
+}
+
```

Comment:
If we use `t.Setenv` we can simply use `os.Getenv` everywhere, at the cost of having the affected tests non-parallel.